x-pack/legacy/plugins/siem/public/containers/matrix_histogram/utils.ts

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { getOr } from 'lodash/fp';
+import { useEffect, useState } from 'react';
+import {
+  MatrixHistogramDataTypes,
+  MatrixHistogramQueryProps,
+} from '../../components/matrix_histogram/types';
+import { DEFAULT_INDEX_KEY } from '../../../common/constants';
+import { useStateToaster } from '../../components/toasters';
+import { errorToToaster } from '../../components/ml/api/error_to_toaster';
+import { useUiSetting$ } from '../../lib/kibana';
+import { createFilter } from '../helpers';
+import { useApolloClient } from '../../utils/apollo_context';
+import { inputsModel } from '../../store';
+import { GetMatrixHistogramQuery, GetNetworkDnsQuery } from '../../graphql/types';
+import * as i18n from './translations';
+
+export const useQuery = <Hit, Aggs, TCache = object>({
+  alertsType = false,
+  anomaliesType = false,
+  authenticationsType = false,
+  dataKey,
+  eventsType = false,
+  endDate,
+  filterQuery,
+  query,
+  stackByField,
+  startDate,
+  sort,
+  isPtrIncluded,
+  isInspected,
+  isHistogram,
+  pagination,
+}: MatrixHistogramQueryProps) => {
+  const [defaultIndex] = useUiSetting$<string[]>(DEFAULT_INDEX_KEY);
+  const [, dispatchToaster] = useStateToaster();
+  const [refetch, setRefetch] = useState<inputsModel.Refetch>();
+  const [loading, setLoading] = useState<boolean>(false);
+  const [data, setData] = useState<MatrixHistogramDataTypes[] | null>(null);
+  const [inspect, setInspect] = useState<inputsModel.InspectQuery | null>(null);
+  const [totalCount, setTotalCount] = useState(-1);
+
+  const getErrorMessage = () => {
```

Comment:
consider passing something like `fetchErrorMessage` as a prop to the component, as that would eliminate the need to maintain this branching logic when new types are added